nix/migration-tests.nix

```diff
@@ -0,0 +1,105 @@
+# Test harness for migrating databases from previous versions to the
+# current version.
+
+{ pkgs ? import ./nixpkgs-haskell.nix  {} }:
+
+with pkgs.lib;
+
+let
+  # List of git revisions to test against.
+  releases = [
+    { rev = "v2019-11-18";
+      sha256 = "0ijsdi8f0y49hn62dj80xipf14nb0hnisddq6b8lj10jznwgsdag";
+      allowFail = true; }
+    { rev = "v2019-12-09";
+      sha256 = "1ld415pbbhj8hg14325w2dcwxkkanm3vx09s7cs1viy8lcl69sln";
+      allowFail = true; }
+    { rev = "v2019-12-13";
+      sha256 = "1a2b4iflwwp824b1k9b82jw2q8pqlar6hg96nv03zv55glkgdllm";
+      allowFail = true; }
+    { rev = "v2019-12-16";
+      sha256 = "0y5xf43lrc7gygvxh5ywkglr3d1xcc19dsskm7frl0v6m9yxzni6"; }
+  ];
```

Comment:
:thinking: I am not sure we have to go that far in the past. In particular, `v2019-11-18` is certainly not compatible with `v2019-12-09`, isn't it ?